tests/DateTimeTest.php

```diff
@@ -345,4 +345,186 @@ public function testToDateTimeImmutable(): void
+            [
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(2018, 11, 13),
+                    new Time(12, 0, 0)
+                ),
+                false,
+            ],
+            [
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(12, 0, 0)
+                ),
+                false,
+            ],
+            [
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(11, 0, 0)
+                ),
+                false,
+            ],
+            [
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(13, 0, 0)
+                ),
+                true,
+            ],
+            [
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(2018, 11, 15),
+                    new Time(12, 0, 0)
+                ),
+                true,
+            ],
+            [
+                new DateTime(
+                    new Date(500, 1, 1),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(2018, 1, 1),
+                    new Time(12, 0, 0)
+                ),
+                true,
+            ],
+            [
+                new DateTime(
+                    new Date(2018, 1, 1),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(500, 1, 1),
+                    new Time(12, 0, 0)
+                ),
+                false,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider isBeforeProvider
+     */
+    public function testIsBefore(DateTime $dateTimeA, DateTime $dateTimeB, bool $expectedResult): void
+    {
+        self::assertSame($expectedResult, $dateTimeA->isBefore($dateTimeB));
+    }
+
+    public function isBeforeProvider(): array
+    {
+        return [
+            [
+                new DateTime(
+                    new Date(2018, 11, 14),
+                    new Time(12, 0, 0)
+                ),
+                new DateTime(
+                    new Date(2018, 11, 13),
+                    new Time(12, 0, 0)
+                ),
+                true,
+            ],
+            [
+                new DateTime(
```

Comment:
extract this case into seperate test case:
one date is enough to provide the results + you can test both methods in that case